Z64Utils/Z64/Z64Object.cs

```diff
@@ -368,119 +420,796 @@ public override void SetData(byte[] data)
+                    VertexListSeg = new SegmentedAddress(br.ReadUInt32());
+                    NbPolygons = br.ReadUInt16();
+                    br.ReadUInt16();
+                    PolyListSeg = new SegmentedAddress(br.ReadUInt32());
+                    SurfaceTypeSeg = new SegmentedAddress(br.ReadUInt32());
+                    CamDataSeg = new SegmentedAddress(br.ReadUInt32());
+                    NbWaterBoxes = br.ReadUInt16();
+                    br.ReadUInt16();
+                    WaterBoxSeg = new SegmentedAddress(br.ReadUInt32());
+                }
+            }
+            public override int GetSize() => COLHEADER_SIZE;
+        }
+        public class CollisionVerticesHolder : ObjectHolder
+        {
+            public const int ENTRY_SIZE = 6;
+
+            public Vec3s[] Points { get; set; }
+
+            public CollisionVerticesHolder(string name, byte[] data) : base(name)
+            {
+                SetData(data);
+            }
+            public override EntryType GetEntryType() => EntryType.CollisionVertices;
+
+            public override byte[] GetData()
+            {
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    BinaryStream bw = new BinaryStream(ms, ByteConverter.Big);
+
+                    for (int i = 0; i < Points.Length; i++)
+                    {
+                        bw.Write(Points[i].X);
+                        bw.Write(Points[i].Y);
+                        bw.Write(Points[i].Z);
+                    }
+
+                    return ms.GetBuffer().Take((int)ms.Length).ToArray();
+                }
+            }
+
+            public override void SetData(byte[] data)
+            {
+                if ((data.Length % ENTRY_SIZE) != 0)
+                    throw new Z64ObjectException($"Invalid data size (0x{data.Length:X}) should be a multiple of 6");
+
+                Points = new Vec3s[data.Length / ENTRY_SIZE];
+                using (MemoryStream ms = new MemoryStream(data))
+                {
+                    BinaryStream br = new BinaryStream(ms, ByteConverter.Big);
+
+                    for (int i = 0; i < Points.Length; i++)
+                        Points[i] = new Vec3s()
+                        {
+                            X = br.ReadInt16(),
+                            Y = br.ReadInt16(),
+                            Z = br.ReadInt16()
+                        };
+                }
+            }
+            public override int GetSize() => Points.Length * ENTRY_SIZE;
+        }
+        public class CollisionPolygonsHolder : ObjectHolder
+        {
+            public const int ENTRY_SIZE = 0x10;
+
+            public struct CollisionPoly
+            {
+                public ushort Type;
+                public ushort[] Data;
+                public Vec3s Normal;
+                public short Dist;
+            }
+
+            public CollisionPoly[] CollisionPolys { get; set; }

```

Comment:
```suggestion
            public Poly[] polygons { get; set; }
```
being in a `CollisionPolygonsHolder` is probably enough to understand it's collision-related

Reply:
Eventually when I resurrect my Collision Renderer progress it will be used outside of the Holder.